### ISSUE TITLE
Request: Folder and file share classification improvement

### ISSUE THREAD

**Issue by OlivierS1** [nextcloud/android#1217]
### Actual behaviour
-Tell us what happens
When folders or files have the same name across accounts, and one or more of those files and folders are shared, the name of the folder of file will be have a "(N)" N being a number chosen by NC (incremental value it seems).
If I have a Document on one account and share it with another account, the name on the second account would be "Document (2)".
From the user who was given access to the share, it is not evident to say the share is from another account, nor from whom the share came from. 
Under NC Dev 20170721, it doesn't even seem possible to know who provided the share.
For security and privacy reasons, and to prevent saving files in the wrong account, it would be really nice if the name of the file or folder containing the name of the person or group the share originated from. Instead of using the "(N)" designation, maybe "(username)" or for a group, the "Groupname".
To prevent such confusion with "Autoupload" default folder name, I usually add "_username" at the folder so I would get "Instantupload_username".
I find this approach makes sharing and receiving shares clearer to read and process.

### Expected behaviour
-Tell us what should happen
 Receiving a file or folder share, the name of the user or group should appear on the shared file or folder.
### Steps to reproduce
1. 
2. 
3. 

### Environment data
Android version:

Device model: 

Stock or customized system:

Nextcloud app version:

Nextcloud server version:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
Please have a look into the web ui, there you will see the user or group name beside the sharing icon.
Would this be sufficient for you? Then we would also have the same look & feel.

![screenshot_20170721_091832](https://user-images.githubusercontent.com/5836855/28453026-95a7fb52-6df5-11e7-92a4-d331638f86cd.png)

**Comment by OlivierS1** [comment]
@tobiasKaminsky One of the major issue I have with shares in general is knowing if the share belongs to me or to others.
As I am looking into this issue, I start learning a few subtlelties about the naming convention of the shares.
In my mind the keyword "Shared" next to the web UI meant the folder or file was shared, it could have been my share or someone sharing it with me.
Now I start to understand that my shares have the keyword "Shared" while the name of the person who is sharing a folder or file with me will have their name next to the shared icon.
Yet, to the new and occasional user, and the more experienced, this is not obvious.
I think we need to brainstorm this idea. I will experiment a bit more with it.
What would be your suggestion in this area?

I would like to consider one possibility, and that is to insert a new line underneath the folder of file name, below or above the line indicating the size and date.

**Comment by tobiasKaminsky** [comment]
I understand your difficulty to figure out these share types. 
![screenshot_20170724_085720](https://user-images.githubusercontent.com/5836855/28511565-1d89570a-704e-11e7-9cec-d55ef4787656.png)
- 1, 1_Photos is shared by me via link
- NewShare is shared by me to an user --> *maybe add the user name/group" (shared with TestUser)?
- sharedWithTobi is shared by admin with me (tobi) --> *shared by admin"?

As we have to stay in sync with webUI, @nextcloud/designers 
A new line might clutter up the whole ui, especially when we add the "three dots" like in the web ui per each line ( #579)

**Comment by jancborchardt** [comment]
@tobiasKaminsky yep, we should do it like in the web interface. However due to space constraints, instead of the full name we should just use the avatar right next to the share icon. (This is also something we should do in the web interface anyway.)

Would that work?

**Comment by jancborchardt** [comment]
I opened a pull request to improve this in the web interface at https://github.com/nextcloud/server/pull/6094 – posting here as mockup:
(Left current, right new design)
![screenshot from 2017-08-13 15-40-11](https://user-images.githubusercontent.com/925062/29250244-5bf53120-803f-11e7-90f3-27bf780b5c25.png)

**Comment by tobiasKaminsky** [comment]
What is the distinction between "shared with jan" and "shared by jan"? As said above I think this must be more visible.

**Comment by jancborchardt** [comment]
There is none at the moment, and from a UX perspective it matters less - what's important is who it's visible by. In the future the owner could have some extra circle around them (and would be sorted first in the list).

**Comment by tobiasKaminsky** [comment]
If it matters less, then we could also show the user/avatar on the "shared with" type, or?

**Comment by jancborchardt** [comment]
> we could also show the user/avatar on the "shared with" type, or?

Yup! As said in the pull request this is only the first step. Actually on all shares, the avatars should be shown (with a limit and overflow as described there).

> who is responsible for the share/who can edit it

That's why I said above:
In the future the owner could have some extra circle around them (and would be sorted first in the list).

:)
